web/src/settings_account.ts

```diff
@@ -751,12 +767,20 @@ export function set_up(): void {
 
     $("#profile-settings").on("change", ".custom_user_field_value", function (this: HTMLElement) {
         const fields: CustomProfileFieldData[] = [];
-        const value = $(this).val()!;
+        let value = $(this).val()!;
         assert(typeof value === "string");
         const field_id = Number.parseInt(
             $(this).closest(".custom_user_field").attr("data-field-id")!,
             10,
         );
+        if ($(this).hasClass("date-field") && !is_valid_date_format(value)) {
+            const converted = parse_to_iso_date_format(value);
+            if (converted) {
+                value = converted;
+            } else {
+                value = "";
```

Comment:
I guess this is for handling invalid strings, but I don't think it is correct as it will remove the value. Value should only be removed if the text input is made empty, but not when input having invalid strings.

I am not sure how to handle invalid strings here, there isn't any button which is clicked to confirm changes which can be disabled. @alya thoughts?

Reply:
> I am not sure how to handle invalid strings here, there isn't any button which is clicked to confirm changes which can be disabled. @alya thoughts?

@alya any feedback on how to handle invalid strings?

Reply:
Oh, sorry to miss the question earlier. We should probably outline the field in red, and revert to the previous value when it loses focus.